docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,282 @@
+externalConfigSecretName: my-mimir-config
+externalConfigVersion: "0"
+```
+
+### Use external Secret
+
+Prepare a `Secret` object where the configuration is base64 encoded and placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: <configuration-base64>
+```
+
+Replace `<configuration-base64>` with the configuration encoded as base64 format string. The name `my-mimir-config` is just an example.
+
+Set the following value for the Helm chart:
+
+```yaml
+useExternalConfig: true
+externalConfigSecretName: my-mimir-config
+configStorageType: Secret
+externalConfigVersion: "0"
+```
+
+### Update the configuration
+
+In order to make components aware of configuration changes, either:
+
+- Update the value in `externalConfigVersion` and run `helm update`
+- or restart components affected by the configuration change manually
+
+## Injecting credentials
+
+Credentials should be kept in `Secret` objects or in a credential vault. The Helm chart value `global.extraEnvFrom` can be used to inject the credentials into the runtime environment variables of the Grafana Mimir components. The data keys will become environment variables and usable in the Grafana Mimir configuration. For example `AWS_SECRET_ACCESS_KEY` can be referenced as `${AWS_SECRET_ACCESS_KEY}` in the configuration. See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
+
+Grafana Mimir will not keep track of changes to the credentials. If the credentials change, Grafana Mimir services should be restarted to use the new value. An easy way to trigger such restart is to provide a `global.podAnnotation` which will be applied to all Grafana Mimir components. Changing the value of the global annotation will instruct Kubernetes to restart the components. For example changing `global.podAnnotations.bucketSecretVersion` from `'0'` to `'1'` triggers a restart - note that [pod annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) can only be strings.
+
+## Example of configuration managed with Helm
+
+This example show how to set up the configuration to use an S3 bucket for blocks storage. We assume that the namespace in use is called `mimir-test`.
+
+1. Set up the external blocks storage, in this case S3 with buckets named for example `my-blocks-bucket`, `my-ruler-bucket` and in case of Grafana Enterprise Metrics `my-admin-bucket`.
+
+1. Create an external secret with the S3 credentials by writing the following to a `mysecret.yaml` file:
+
+   ```yaml
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: mimir-bucket-secret
+   data:
+     AWS_ACCESS_KEY_ID: FAKEACCESSKEY
+     AWS_SECRET_ACCESS_KEY: FAKESECRETKEY
+   ```
+
+   Replace FAKEACCESSKEY and FAKESECRETKEY with the actual value encoded with base64.
+
+1. Apply the secret to your cluster with the `kubectl` command:
+
+   ```bash
+   kubectl -n mimir-test apply -f `mysecret.yaml`
```

Comment:
```suggestion
   kubectl -n mimir-test apply -f mysecret.yaml
```

these ticks have a [meaning in bash](https://unix.stackexchange.com/questions/27428/what-does-backquote-backtick-mean-in-commands/27432#27432)